core/src/main/java/net/consensys/eventeum/endpoint/TransactionMonitoringEndpoint.java

```diff
@@ -27,21 +24,14 @@
     /**
      * Monitors a transaction with the specified hash, on a specific node
      *
-     * @param identifier the transaction identifier (hash for now)
-     * @param nodeName the name of the node that should be monitored for the transaction
+     * @param TransactionMonitoringSpec the transaction spec to add
      * @param response the http response
      */
     @RequestMapping(method = RequestMethod.POST)
-    public MonitorTransactionsResponse monitorTransactions(@RequestParam String identifier,
-                                                           @RequestParam(required = false) String nodeName,
+    public MonitorTransactionsResponse monitorTransactions(@RequestBody TransactionMonitoringSpec spec,
                                                            HttpServletResponse response) {
-
-        if (nodeName == null) {
-            nodeName = Constants.DEFAULT_NODE_NAME;
-        }
-
-        final TransactionMonitoringSpec spec =
-                new TransactionMonitoringSpec(TransactionIdentifierType.HASH, identifier, nodeName);
+        spec.generateId();
```

Comment:
Main doubt that I had is here. I leave it like this, but glad to change it. The problem is that the `@requestBody` create the entity with setters and not with the construct. So our entity is not with the fields as we want. Since the id is created hashing the other params.

Options:

- Create a `TransactionMonitoringUtils.generateId(spec)` that receives the object and returns a new one trying to avoid side effects.
- Create a dto to get the request and then call the `TransactionMonitoringSpec` construct with the dto that construct the proper model.

But since I was not sure which one you prefer, I added this which is in the middle.